api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters-blm/asset-factory.service.ts

```diff
@@ -17,15 +22,43 @@ export class AssetFactory {
 
     copySync(from.workingDirectory, to.workingDirectory, {
       // Marxan binary is already linked
-      filter: (src, dest) => !src.match(/marxan/),
+      filter: (src) => !src.match(/marxan/),
     });
 
-    // TODO modify input.dat file - replace "BLM" with overrideBlmValue
+    const inputDat = this.getInputDat(assets);
+    if (!inputDat) throw new Error('No URL for input.dat found');
+
+    const inputDatPath = resolve(
+      to.workingDirectory,
+      inputDat.relativeDestination,
+    );
+
+    await this.ensureWriteDirectoryExists(inputDatPath);
+
+    const input = this.reader.read(inputDatPath);
+
+    // TODO: Why the input.dat is coming without BLM?
+    // const matcher = new RegExp(/BLM\s.*/);
+    const inputWithCorrectBLM = `BLM ${overrideBlmValue}\n${input}`;
+    /*input.replace(
+      matcher,
+      `BLM ${overrideBlmValue}`,
+    );*/
```

Comment:
```suggestion
    const inputWithCorrectBLM = `BLM ${overrideBlmValue}\n${input}`;
```

Reply:
I've renamed the variables and added a method for checking if the file already contains BLM parameter, if it is present in the file it is replaced, in case that the file doesn't contain the parameter it is appended at the end of the file